### ISSUE TITLE
EPIC: New Machine requirement: Windows dockerBuild containers

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#3286]
I need to request a new machine:

- New machine operating system (e.g. linux/windows/macos/solaris/aix): Windows
- New machine architecture (e.g. x64/aarch32/arm32/ppc64/ppc64le/sparc): x64
- Provider (leave blank if it does not matter): Docker :-)
- Desired usage: Build containers, similar to what we have for Linux
- Any unusual specification/setup required:
- How many of them are required: n/a - they should be created dynamically

Please explain what this machine is needed for:
Running builds in an isolated way where we can achieve SLSA build level 3 compliance on Windows along with the other primary platforms. Ideally we'll be able to create windows-on-windows container images which we share and then download and run the builds in. 

As background info:
- https://learn.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=dockerce#windows-server-1 looks to be a good resource on the software options for containers on Windows
- Eclipse Temurin does provide container images for Windows at https://hub.docker.com/_/eclipse-temurin/ - these are created from the dockerfiles in https://github.com/adoptium/containers/tree/main/21/jdk/windows which may give a good starting point for this work on investigating windows containers.

So the tasks required would be:
- [x] Identify the appropriate software for running containers and ensure no licensing concerns (Likely something from the microsoft site linked above)
- [x] See if we can verify that a "basic" dockerfile works in that environment and whether we can map directories into it (same as `-v` on linux) which are read+write in the container
- [x] Determine whether we can create a container from the playbooks using a dockerfile equivalent to [the Linux ones](https://github.com/adoptium/infrastructure/tree/master/ansible/docker)
- [x] Once we create the container, map a directoryi from the host into it with `-v` and use that to build Temurin in the container on the mapped volume so that the output is visible on the host system.
- [x] Understand whether we can reasonably push the resulting container images with the compiler up to dockerhub _[Answer: We will push them to gcr.io and mirror to azurecr for better performance on azure dynamic machines.]_
- [x] Integrate this into [the build pipelines](https://github.com/adoptium/ci-jenkins-pipelines/blob/7b9559ce88321ff8111180fdc58421a0f9eadcef/pipelines/build/common/openjdk_build_pipeline.groovy#L1876)
- [x] Implement processes to regenerate the images when playbook updates are made, - likely an addition to what we do for Linux in https://github.com/adoptium/infrastructure/blob/master/FAQ.md#what-about-the-builds-that-use-the-dockerbuild-tag
- [x] Declare SLSA Build level 3 on Windows :-)

Once this level of analysis and expertise is gained it will likely make windows installer testing, or any other such activities simpler and give us more options moving forward.

Related for historic reference:
- https://github.com/adoptium/infrastructure/pull/2440
- https://github.com/adoptium/infrastructure/pull/2556

**Comment by RadekCap** [comment]
Please, assign this task to me. Thank you.

**Comment by sxa** [comment]
Of the three options listed on the Microsoft website:
- The first (Docker CE / Moby) seems to work well out of the box
- The second (Mirantis) appears to be a commercial offering
- The third (Containerd+nerdctl) appers functional although networking doesn't work out of the box and it seems to fail to be able to start the eclipse-temurin container's default `jshell` process.

**Comment by sxa** [comment]
OK First phase done ...
- `docker run -p 5986:5986 -v c:\Users\sxa:c:\cabbage mcr.microsoft.com/windows/servercore:ltsc2022`
- Run [ConfigureRemotingForAnsible.ps1](https://raw.githubusercontent.com/ansible/ansible-documentation/devel/examples/scripts/ConfigureRemotingForAnsible.ps1) with the [usual parameters](https://github.com/adoptium/infrastructure/blob/7e74e63a0b19ca431bb4fbcbe388260a8c92619f/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml#L16) with the netsh commands disabled (They require windows defender which isn't in the image)
- Create a user to connect with for the playbooks:
```
net user ansible MyPassword /ADD
net localgroup "Administrators" ansible /ADD
net localgroup "Remote Management Users" ansible /ADD
```
This allows the machine to be accessible via ansible running on a remote machine :-)

**Comment by sxa** [comment]
Playbook execution notes:
- VS2013 requires the archive under `/Vendor_Files/windows`, otherwise `MSVS_2013` needs to be skipped
- `NTP_TIME` needs to be skipped as that has issues that are presumably related to running in a container: `FAILED! => {"changed": false, "msg": "Unhandled exception while executing module: Service 'Windows Time (W32Time)' cannot be started due to the following error: Cannot start service W32Time on computer '.'."}`
- In the absence of the fixed layout files for [VS2019](https://github.com/adoptium/infrastructure/blob/4aa7788325c224484f99aa1ae000f117e9b081d7/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2019/tasks/main.yml#L75) and [VS2022](https://github.com/adoptium/infrastructure/blob/4aa7788325c224484f99aa1ae000f117e9b081d7/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022/tasks/main.yml#L101), `adoptopenjdk` needs to be skipped to allow them to complete successfully

**Comment by sxa** [comment]
ansible can be run on the host to point at the container if you install cygwin which has `ansible` as one of its installable options (You probably want to include `git` too if it's a clean install on the host system). Noting that if you use `localhost`/`127.0.0.1` in your hosts file you should specify `-e git_sha=12345` or something appropriate otherwise the execution will trip up over https://github.com/adoptium/infrastructure/blob/4aa7788325c224484f99aa1ae000f117e9b081d7/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/logs/tasks/main.yml#L14

**Comment by sxa** [comment]
Latest attempt is with:
`--skip-tags adoptopenjdk,reboot,MSVS_2017,NTP_TIME`
Playbook changes to make it complete:
- Set ansible_connection/ansible_winrm_transport  in `ansible.cfg`
- Set ansible_user/ansible_password in `group_vars/all/adoptopenjdk_variables.yml`
- Remove `win_reboot:` from Common/roles/main.yml Line 60
- Remove `win_reboot:` from MSVS_2013 role line 50
- Remove `win_reboot:` from MSVS_2017 role line 37
- Remove `checksum` parameters MSVS_2022 role line 103 as it's been updated
- Remove `win_reboot` from WMF_5.1 role line 29
- Remove `win_reboot` from cygwin role line 45 (Although it's already covered with th `reboot` tag)

**Comment by sxa** [comment]
`docker commit` didn't work on my image:
`Error response from daemon: re-exec error: exit status 1: output: mkdir \\?\C:\Windows\SystemTemp\hcs376450290\Files: Access is denied`
This is specific to the new image which has had the playbook run on it  and does not occur when attempting to commit a image with only basic changes applied.

**Comment by sxa** [comment]
This is unfortuante. The builds aren't working because it looks like the [automatic shortname generation](https://github.com/adoptium/infrastructure/blob/490dcbac85b0246da0968474e5887f20a3c200c5/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml#L27)  (`fsutil behavior set disable8.3 0`) does not appear to be working within the container but is mandatory for the openjdk build process. Directories can have a shortname created manually with `fsutil file setshortname "Long name" shortname` but that is not ideal to do for each possible path.

**Comment by sxa** [comment]
Manually created a few of the shortnames that the configure step was objecting to and I have a JDK21u build complete in a container, so this seems feasible 👍🏻

**Comment by sxa** [comment]
Noting that we should look at doing this with the MS build tools installer which is [suitable for use by Open Source projects](https://devblogs.microsoft.com/cppblog/updates-to-visual-studio-build-tools-license-for-c-and-cpp-open-source-projects/). The jdk21u builds currently use:
```
10:04:20  * C Compiler:     Version 19.37.32822 (at /cygdrive/c/progra~1/micros~3/2022/commun~1/vc/tools/msvc/1437~1.328/bin/hostx64/x64/cl.exe)
10:04:20  * C++ Compiler:   Version 19.37.32822 (at /cygdrive/c/progra~1/micros~3/2022/commun~1/vc/tools/msvc/1437~1.328/bin/hostx64/x64/cl.exe)
```
Other references (this numbering is more confiusing that I realised - I thought we only had the '2022' vs '19.xx' versioning differences to worry about before today...)
- https://devblogs.microsoft.com/oldnewthing/20221219-00/?p=107601 (Would suggest, extrapolating from the table, that _MSC_VER 1937 maps to VS 17.7)
- https://learn.microsoft.com/en-us/cpp/overview/compiler-versions?view=msvc-170#service-releases-starting-with-visual-studio-2017

**Comment by sxa** [comment]
Struggling with the GPG role at the moment which is called during the ANT role (I'm getting `gnupg` as a requirement which supplies `gpg2` instead of `gpg`). Also `Wix` has to be skipped as I don't have `ansible.builtin.runs` available.

Other than that a two-phase dockerfile is looking quite promising. The first sets up WinRM (will only be invoved locally) and installs cygwin with git and ansible, then triggers a reboot to ensure the cygwin path takes effect.

The second runs the playbooks as normal, although for now I've currently it running in multiple layers for performance of testing to allow the caching of each layer to take effect independently:
1. `--skip-tags adoptopenjdk,reboot,ANT,NTP_TIME,Wix,MSVS_2013,MSVS_2017,MSVS_2019,MSVS_2022`
2. `-t ANT`
3. `-t MSVS_2019`
4. `-t MSVS_2022`

This is currently using the playbook branch at https://github.com/sxa/infrastructure/tree/sxa_allhosts which makes a few changes to support this execution.

**Comment by sxa** [comment]
The above approach seemed to work yesterday now that the machine is rebooted after adding cygwin to the PATH and I had a system which was able to successfully build jdk21u using two dockerfiles (First to configure WinRM, the second to run the playbooks using the individual layers from the previous comment. Next steps as follows:

- Verify that on a clean image (I made some changes inside the image after my infrastructure branch was extracted, so that needs to be confirmed as captured in the branch) 
- Fix Wix install
- Fix the git_sha detection
- Update the MSVS_2022 role to use MS build tools to ensure reproducibility of the builds
- Ideally test with the MSVS_2013 and 2017 installers available in the image so those roles do not need to be skipped.

**Comment by sxa** [comment]
Now fixed the path setting so that it only requires one dockerfile so we have something consistent with what we have on Linux now 👍🏻

**Comment by sxa** [comment]
VS2013 install appears to complete OK (Based on the logs in `C:\Windows\SystemTemp` - more detailed logs are in `C:\Temp`) but the playbook doesn't terminate that role so it never continues.

**Comment by sxa** [comment]
Steps to set up:
- Provision Windows Server 2022 machine - ideally with at least 250GB on the C: file system
- [Install cygwin](https://www.cygwin.com/install.html) with `git` and `ansible` support added
- (Ideally set up a second disk for Docker and set the data root there, but I'm not certain how I did that)
- Set 8.3 name support: `fsutil 8dot3name set 0` (Otherwise shortnames can't be set within the containers which cygwin will need to run our automation)
- Install docker either with the [helper script](https://learn.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=dockerce#windows-server-1) (`Invoke-WebRequest -UseBasicParsing "https://raw.githubusercontent.com/microsoft/Windows-Containers/Main/helpful_tools/Install-DockerCE/install-docker-ce.ps1" -o install-docker-ce.ps1` then `.\install-docker-ce.ps1` or with the manual download steps at https://docs.docker.com/engine/install/binaries/#install-server-and-client-binaries-on-windows
- Run `docker run mcr.microsoft.com/windows/servercore:ltsc2022 cmd /c dir /x` to verify that there is a visible shortname for the `Program Files` and `Program Files (x86)` directories.
- Grab [Dockerfile.win2022](https://raw.githubusercontent.com/sxa/infrastructure/windows_docker_fixes/ansible/docker/Dockerfile.win2022) from the infrastructure repo and run `docker build --build-arg PW=Some-Pa55wd -t win2022_build_image -f Dockerfile.win2022 2>&1 | \cygwin64\bin\tee ansible.log` (The PW parameter doesn't matter as long as it's valid for a windows user as it only exists during the ansible run)

**Comment by sxa** [comment]
Based on https://github.com/adoptium/temurin-build/issues/2922#issuecomment-2269480488 we may be able to switch to using Visual Studio 2022 for everything which would significantly reduce the windows installation requirements. The dockerfile is currently set up to only install VS2022 and not the other versions.

**Comment by sxa** [comment]
Next bullet on the list is to: Integrate this into [the build pipelines](https://github.com/adoptium/ci-jenkins-pipelines/blob/7b9559ce88321ff8111180fdc58421a0f9eadcef/pipelines/build/common/openjdk_build_pipeline.groovy#L1876)

Initial attempts using a jenkins workspace directory with a drive on `F:` failed because the jenkins docker failed to map it into `F:` in the container as there was only a `C:` drive. Switched the workspace directory to `C:\jenkins-workspace` and we hit path limits:
```
10:03:48  configure: error: Your base path is too long. It is 112 characters long, but only 100 is supported
```
Now moved to using `C:\ws` for the directory and it seems to be progressing well:
Machine: [dockerhost-azure-win2022-x64-1](https://ci.adoptium.net/computer/dockerhost%2Dazure%2Dwin2022%2Dx64%2D1/) (temporary, called `sxa-win2022-3` in the Azure console)
Build job: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-windows-x64-docker/

Noting that I have had errors like this and while I have not identified the exact cause, clearing out the `build-scripts` directory in the workspace resolves it:
```
10:24:33  Checked out HEAD commit SHA:
[Pipeline] sh
10:24:34  sh: c:/jw/workspace/build-scripts/jobs/jdk21u/jdk21u-windows-x64-docker@tmp/durable-34ace7f2/script.sh.copy: No such file or directory`
```

**Comment by sxa** [comment]
Job https://ci.adoptium.net/job/win2022_docker_image_updater is being prototyped to create the docker image. It is a stripped down copy of the rhel7/s390x one and will save to `win2022_notrhel_image` on the host for now, and as per earlier comments it does not include the infrastructure SHA.

**Comment by sxa** [comment]
Note: The HOME environment variable set when the jenkins agent is started is significant, as it affects where git picks up the `.gitconfig` from during the pipeline checkout on the host. On the current machine I'm using for testing this is set in the `startjenkins.sh` script before the agent is started.

**Comment by sxa** [comment]
Above PR should fix the issue with long file names - I'm doing some extra tests to verify with my current job and have also initiated https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-windows-x64-temurin/138/console to test with the full job name. It should be good with the PR in place as it's using the same logic for overriding the default workspace location as we use in the non-docker situation on Windows.

**Comment by sxa** [comment]
For my own reference - the build times on the docker machine (Not as powerful as the main build machines - it's 2 core / 8GiB) are:

Version | Time for build
---|---
jdk8u | 52m (31m on main build machine)
jdk11u | (1h31 on main build machine)
jdk17u | (1h27 on main build machine)
jdk21u | (1h29 on main build machine)
jdk24 | (1h45 on main build machine)

**Comment by sxa** [comment]
First build using the main pipelines on the dockerhost machine: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-windows-x64-temurin/151/

**Comment by sxa** [comment]
It's been quite a lot of work but the sign_Verification job now has a working run after a refactor of the code that does the signing and assembly within the pipelines. Ref: https://github.com/adoptium/infrastructure/issues/3709#issuecomment-2373386390
A bit of cleaning up, and then verifying that it can create reproducible builds, will mean this can go in as a PR.

**Comment by sxa** [comment]
`--create-sbom` not currently working as `ant` is not in the PATH on the machine.

**Comment by sxa** [comment]
Noting that when attempting to run a build using a fixed SCM_REF for a reproducibility comparison some problems occur

The create_installer_windows job needs to have the `PRODUCT_*_VERSION` fields matching the directory layout for the build. As an example [windbld#986](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/windbld/986/parameters/) which was built with an `SCM_REF` of `jdk-21.0.4+7_adopt` [produced a zip file](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/windbld/986/artifact/workspace/target/OpenJDK21U-jdk_x64_windows_hotspot_2024-10-11-17-35.zip) with a top level directory of `jdk-21.0.5+9` but the JDK inside has `OpenJDK Runtime Environment Temurin-21.0.4+7-202410111909 (build 21.0.4-beta+7-202410111909` in the `java -version` output. This causes the installer job to baulk at the end of a loop searching for a path name with 21.0.4 in it (these lines are not consecutive but there's a lot of debug stuff in these logs). Once it fails to find something it shows the directoriy it has which clearly has an unexpected version number in it.
```
looking for .\SourceDir\OpenJDK21\hotspot\x64\jdk-21.0.4
looking for .\SourceDir\OpenJDK21\hotspot\x64\jdk21u4-b7
looking for .\SourceDir\OpenJDK21\hotspot\x64\jdk-21+7
looking for .\SourceDir\OpenJDK21\hotspot\x64\jdk-21.0.4+7
looking for .\SourceDir\OpenJDK21\hotspot\x64\jdk-21.0.4.0+7
looking for .\SourceDir\OpenJDK-Latest\hotspot\x64\jdk-21.0.4+7
SOURCE Dir not found / failed
Listing directory :
F:\workspace\workspace\build-scripts\release\create_installer_windows\wix\SourceDir\OpenJDK21
F:\workspace\workspace\build-scripts\release\create_installer_windows\wix\SourceDir\OpenJDK21\hotspot
F:\workspace\workspace\build-scripts\release\create_installer_windows\wix\SourceDir\OpenJDK21\hotspot\x64
F:\workspace\workspace\build-scripts\release\create_installer_windows\wix\SourceDir\OpenJDK21\hotspot\x64\jdk-21.0.5+9
```
This is likely nothing to do with the docker changes, but is likely something we should look to address as a build issue in the general case when building something that isn't the latest version, such as a previous GA level. (FYI @andrew-m-leonard)

**Comment by sxa** [comment]
Reproducibility tests based on 21.0.5+11:

Job | docker | UCRT params | devkit param | Result
---|---|---|---|---
1065 | ✅ | ✅ | ❌ | Differences in jmods
1066 | ❌ | ✅ | ❌ | ReproduciblePercent = 100 %
1067 | ✅ | ❌ | ✅ | ReproduciblePercent = 100 %

**Comment by sxa** [comment]
Reproducibility is confirmed good after a number of tests.

A couple of additional notes on this:
- https://lippertmarkus.com/2021/09/04/containers-without-docker-desktop/ may be another option for installing docker but I have not tried this
- The docker commands from the CE version I've used require need to be run as administrator - there is no way to have this run as a normal users (the jenkins one we typically use for the agent does not generally run as an administrative user)

**Comment by sxa** [comment]
New machines being tested:
- [dockerhost-azure-win2022-x64-1-intel](https://ci.adoptium.net/computer/dockerhost%2Dazure%2Dwin2022%2Dx64%2D1%2Dintel/) - 2 core Xeon 8171M/8GiB (B2ms) - 128GiB C: and 128GiB F: for docker (although it's not mapped in)
- [dockerhost-azure-win2022-x64-2-amd](https://ci.adoptium.net/computer/dockerhost-azure-win2022-x64-2-amd) - 2 core AMD EPYC 7763/4GiB (B2als_v2) - 256GiB C: drive and 128GiB F: for Docker
Both machines have had the image created on them, and are able to run builds

The AMD machine completed a jdk21u build in a container from the command line in just under 3h so it is possible to build with 4GiB of RAM. It was slightly slower than the numbers from the B2ms systems in the [earlier comment](https://github.com/adoptium/infrastructure/issues/3286#issuecomment-2305306371), but it's also a different CPU, plus those tests were done with the machine having been worked a bit from the start so may have been subject to bursting limits. Since the AMD one seems to work I will also look at loading up its 256GiB `C:` drive with a normal playbook run (excluding VS2013, 2017 and 2019) so it can act as a drop-in replacement for an existing build machine even without enabling the containerised builds.  The first two here are my prototype machines which will be replaced by the two new ones, but here is a spec comparison so we have the info stored:

Machine | Cores | Docker Disk | RAM | jdk8u | jdk21u | jdk24
---|---|---|---|---|---|---
dh-w22-1 | Xeon 8370C | 250GiB Premium SSD v2 | 8GiB | 
dh-w22-sxa1/3 |  Xeon 8370C | 200GiB Premium SSD | 8GiB |
dh-w22-1-intel | Xeon 8171M | 128GiB HDD | 8GiB | 
dh-w22-2-amd | AMD EPYC 7763 | 128GiB HDD | 4GiB |

**Comment by sxa** [comment]
I have got a "static" container running a jenkins agents which I'm going to use on a burstable machine to test the process with the ea pipelines. This will mean we can switch to/from this performance testing without modifying the pipelines (i.e. the new explicit docker support is not yet being enabled). A couple of notes on this:
1. Other than the first build, this is likely to be quite slow due to the use of burstable machines
2. The `wix` label has been removed from these agents as there is a problem with the locales in the containers (Potentially we could install enough to make WIX work on the docker host system) [*]
3. I have removed the `build` label from the two existing machines so that they will not be used for the builds, but they are kept online so that the installer jobs requiring `wix` can run on them.

[*] - Sample failure: https://ci.adoptium.net/job/build-scripts/job/release/job/create_installer_windows/1108/console
```
Building setup translation for culture "de-de" with LangID "1031"...
Input Error: Can not find script file "C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x64\WiLangId.vbs".
WiLangId failed with : 1
Failed to generate setup translation of culture "de-de" with LangID "1031".
failed to build translation de-de 1031
```

**Comment by sxa** [comment]
I've done a bit of rebasing to allow us to use the updated playbooks (Currently the dockerfile is still pointing at my original fork/branch of the infrastructure repo, which has resulted in it not having recent updates [such as ant](https://github.com/adoptium/temurin-build/issues/4062#issuecomment-2505991477))

I still have to have fixes for ant-contrib (The download isn't working - I have to pull it from a copy I've put in place) and https://github.com/adoptium/infrastructure/pull/3828 is preventing parts of the playbook from completing too with the latest versions.

**Comment by sxa** [comment]
Ref the WiX error [above](https://github.com/adoptium/infrastructure/issues/3286#issuecomment-2504207097), I have done the following to allow it to run on the 8GiB Intel machine:
- Run the playbooks on the HOST excluding  (Although it should only need the WiX role, which requires the .net SDK to be installed correctly first (It didn't install itself properly the first time)
- Created a `jenkins` user
- Started a new agent [dockerhost-azure-win2022-x64-2j](https://ci.adoptium.net/computer/dockerhost%2Dazure%2Dwin2022%2Dx64%2D2j) (Note the `j` suffix) which is labelled to run `wix` jobs only.
Sample run on this agent: https://ci.adoptium.net/job/build-scripts/job/release/job/create_installer_windows/1128/
This means the machine can technically run jobs from three different jenkins agents:
- The dockerhost system that can instantiate containers and run jobs (Running as an administrator)
- The agent that is running in a static container while we transition over to docker (Runs as `ContainerUser` in the contianer.
- The new `jenkins` agent which can run the create_installer_windows jobs.
The first two should not be enabled in parallel (same for the other similar machines) as this will overwhelm them.

**Comment by sxa** [comment]
First pass with JDK8

Machine | Time |
---|---
build-docker-1-amd (on dh-2-amd) | [1h06](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/534/)
build-docker-2-intel (on dh-1-intel) | [1h32](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/535/)
build-docker-3-intel (on dh-2-intel) | [1h40](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/536/)

Note that 3-intel was running some installer creation tests at the same time

With jdk21:

Machine | Time | Reproducible? | Notes
---|---|---|---
b-d-2-intel | | 
b-d-1-amd | |
b-d- 3-intel | |
dh-1-intel | |
dh-2-amd | |
dh-3-intel | |

**Comment by sxa** [comment]
Noting that all -ea builds from this week were run int he static docker containers, and are therefore the first set to be built with machines that only have the MS VS2022 Build Tools installation which meet the requirement in https://github.com/adoptium/temurin-build/issues/3787

**Comment by sxa** [comment]
I've provisioned another non-burstable spot-provisioned 2-core 8GiB system dockerhost-azure-win2022--intel to run some `sanity.openjdk` tests in static containers on the hosts:

Machine | Grinder | Time | Results
---|---|---|---
bd-4-intel | [11993](https://ci.adoptium.net/view/Test_grinder/job/Grinder/11993) | 2h10
bd-3-amd | [11997](https://ci.adoptium.net/view/Test_grinder/job/Grinder/11997/) | |
test-auzre-4 | [11998](https://ci.adoptium.net/view/Test_grinder/job/Grinder/11998/) | |
test-azure-2 | [11999](https://ci.adoptium.net/view/Test_grinder/job/Grinder/11999/) | |

**Comment by adamfarley** [comment]
Hi @sxa - I'm seeing an issue on [dockerhost-azure-win2022-x64-3-intel](https://ci.adoptium.net/computer/dockerhost-azure-win2022-x64-3-intel):
```
15:20:06  C:\workspace\openjdk-build>git clean -fdx 
15:20:06  warning: failed to remove test/: Permission denied
```
This issue is preventing builds from working on that machine.
https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-windows-x86-32-temurin/376/

**Comment by adamfarley** [comment]
Git clean continues to fail.

```
22:17:05  C:\workspace\openjdk-build>git clean -fdx 
22:17:05  warning: failed to remove workspace/: Permission denied
```

Examples:
https://ci.adoptium.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk8u/job/jdk8u-windows-x64-temurin/212/console
https://ci.adoptium.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk11u/job/jdk11u-windows-x64-temurin/220/console
https://ci.adoptium.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk17u/job/jdk17u-windows-x64-temurin/214/console

THE [jdk 21 build](https://ci.adoptium.net/job/build-scripts-pr-tester/job/build-test/job/jobs/job/jdk21u/job/jdk21u-windows-x64-temurin/75/console) didn't see this issue, but we did have rm failures.

```
00:02:59  rm: cannot remove 'C:/workspace/openjdk-build/CONTRIBUTING.md': Permission denied
00:02:59  rm: cannot remove 'C:/workspace/openjdk-build/FAQ.md': Permission denied
00:02:59  rm: cannot remove 'C:/workspace/openjdk-build/LICENSE': Permission denied
```

@sxa - Is this the right place for these bug reports? Maybe you'd prefer separate issues?

**Comment by sxa** [comment]
This is being discussed in slack. It's specific to the PR tester which is using `CLEAN_WORKSPACE_AFTER` which the EA builds do not. (I'm not certain why the x86-32 build two comments up hit it though)

**Comment by adamfarley** [comment]
> (I'm not certain why the x86-32 build two comments up hit it though)

The 32 bit one was a week ago. Maybe it's fixed now?

**Comment by sxa** [comment]
Closing this since the [builds are now happening in containers](https://github.com/adoptium/ci-jenkins-pipelines/pull/1155) which are [being pulled from the Azure registry](https://github.com/adoptium/ci-jenkins-pipelines/pull/1170).
Noting that https://github.com/adoptium/infrastructure/issues/3712 is still open regarding capturing the dockerhost setup specifically in a playbook.
